### DISCUSSION TITLE
SetDefaults and nested config

### DISCUSSION THREAD

**Discussion by mieliespoor** [spf13/viper#1559] [Q&A]
I'm not sure I correctly understand how to set defaults for a nested config.

Take this:

```
type Global struct {
	LogLevel               string
	LogFormat              string
	MetricsPath            string
	ServicePort            string
	EnabledMetrics         PrometheusMetrics
}

type PrometheusMetrics struct {
	projectInfo        bool
	vulnerabilities    bool
	policyViolations   bool
	lastBOMImport      bool
	inheritedRiskScore bool
}
```

I have tried `viper.SetDefault("EnabledMetrics.projectInfo", true)` and also:

```
viper.SetDefault("EnabledMetrics", map[string]bool{
		"projectInfo":        true,
		"inheritedRiskScore": true,
		"lastBOMImport":      true,
		"policyViolations":   true,
		"vulnerabilities":    true,
	})

var cfg Global
err = viper.Unmarshal(&cfg)
```

Neither of the two work, and when I Unmarshal the config, everything under EnabledMetrics is false. Where do I go wrong?

**Comment by cam-schultz** [comment]
Did you ever figure out how to achieve this? Running into the exact same issue with an `int` field in a nested struct that is unmarshalled, for which I'd like to set the default value to `-1` instead of the type's zero value.

**Reply by mieliespoor** [reply]
Unfortunately, not.
